### ISSUE TITLE
WCAG: Warnings on our radiogroup

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#753]
### Description of the bug

According to our testing tools we have issues with the way our radiogroups are set up. 

Wave gives us this warning:
![Skjermbilde 2022-12-15 kl  08 46 36](https://user-images.githubusercontent.com/33222679/207802246-c3087fd4-bde6-4320-ad6f-1892d03ac842.png)

And Siteimprove gives this warning:
![Skjermbilde 2022-12-15 kl  08 47 16](https://user-images.githubusercontent.com/33222679/207802386-7f1f40d1-4f89-4914-80e5-10276b3c6480.png)

This is how it is structured today
![Skjermbilde 2022-12-15 kl  08 49 19](https://user-images.githubusercontent.com/33222679/207802775-7b7be286-ddb8-4981-aaa7-a9abaafd97d9.png)

This should also cover the last task in #14 

Our two issues that I am guessing are causing the warnings
- [ ] There is a `label` inside the legend. The legend should act as the "label" in groups like this.
- [ ] We use a aria role: `radiogroup`. If we set up a group of radios correctly with native HTML this should be unnecessary. [Source and examples](https://developer.mozilla.org/en-US/docs/web/accessibility/aria/roles/radiogroup_role) 
- [ ] Check if we have the same issues on our checkbox group

### Steps To Reproduce

Open an app with a radiogroup and run accessibility check on it.
